src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -595,6 +598,17 @@ let name_loop_vars ctx =
   let ctx = { ctx with loop_level = ll + 1 } in
   match ll with 0 -> (string "loop_vars", ctx) | _ -> (string "loop_vars_" ^^ string (string_of_int ll), ctx)
 
+let prepend_monad exp doc =
```

Comment:
I think it would be good to use this function in `doc_funcl_init` since it's mostly taken from there

Reply:
The difference is that there we extract the `.run` identifiers, not the monad *types*